MariaCraciun1_Task2/DynamicStringArray.cpp

```diff
@@ -0,0 +1,84 @@
+    #include "DynamicStringArray.hpp"
+
+
+    DynamicStringArray::DynamicStringArray(){
+        size=0;
+        dynamicArray=nullptr;
+    }
+    
+    DynamicStringArray::DynamicStringArray(const DynamicStringArray& obj){
+        size=obj.getLength();
+        dynamicArray= new std::string[size];
+        for(int i=0;i<size;i++){
+            dynamicArray[i]=obj.dynamicArray[i];
+        }
+    }
+    
+    DynamicStringArray &DynamicStringArray::operator=(const DynamicStringArray &obj){
+        size= obj.getLength();
+        dynamicArray= new std::string[size];
```

Comment:
What happen here when you have self assignment?

Reply:
That case was not treated. I have fixed this by putting everything, except the return statement in this if :
  if ( this != &obj )
This way if there is self assignment, the current object will be returned.